src/schema-generation/order-by-and-pagination-augmentation.ts

```diff
@@ -91,12 +109,50 @@ export class OrderByAndPaginationAugmentation {
                     description: `The number of items to include in the result. If omitted, all remaining items will be included (which can cause performance problems on large collections).`,
                 },
             },
+            transformResult: (data: any, args: object, executionOptions: ExecutionOptions) => {
+                if (
+                    options?.firstLimitCheckType !== LimitTypeCheckType.RESOLVER ||
+                    'first' in args
+                ) {
+                    return data;
+                }
+                if (
+                    executionOptions.implicitLimitForRootEntityQueries &&
+                    data.length > executionOptions.implicitLimitForRootEntityQueries
+                ) {
+                    throw new Error(
+                        `Collection is truncated by default to ${executionOptions.implicitLimitForRootEntityQueries} elements but contains more elements than this limit. Specify a limit manually to retrieve all elements of the collection.`,
+                    );
+                }
+                return data;
+            },
             resolve: (sourceNode, args, info) => {
                 let listNode = schemaField.resolve(sourceNode, args, info);
                 let itemVariable = new VariableQueryNode(decapitalize(type.name));
                 let objectNode = this.rootFieldHelper.getRealItemNode(itemVariable, info);
 
-                const maxCount: number | undefined = args[FIRST_ARG];
+                let maxCount: number | undefined = args[FIRST_ARG];
+
+                if (
+                    options?.firstLimitCheckType &&
+                    info?.maxLimitForRootEntityQueries &&
```

Comment:
```suggestion
                    info.maxLimitForRootEntityQueries &&
```

`info` is not optional.